### PR TITLE
Define and use a node record for erlpmd_ets

### PR DESCRIPTION
**Pull request by puzza007**
Untested!

Easier to do matches etc

### CODE REVIEW

File: src/erlpmd_ets.erl [hunk 1]
```diff
@@ -36,19 +42,20 @@ node_port(NodeName, erlpmd) ->
...
 dump(all, erlpmd) -> dump('_', erlpmd);
 dump(NodeType, erlpmd) ->
     {ok, [{Name, Port, Fd}
           || [Name, Port, Fd] <- ets:match(erlpmd,
-                                           {'$1', {'$2', NodeType, '_', '_', '_', '_', '$3', '_'}})]}.
+                                           #node{name='$1', port_no='$2', type=NodeType, fd='$3', _='_'})]}.
 
 node_stopped(Fd, erlpmd) ->
-	case ets:match(erlpmd, {'$1', {'_', '_', '_', '_', '_', '_', '_', Fd}}) of
+	case ets:match(erlpmd, #node{name='$1', fd=Fd, _='_'}) of
```
Comment (puzza007) [inline src/erlpmd_ets.erl:54]: I think this might have been a bug. `Fd` is second last when the code is inserted.

File: src/erlpmd_ets.erl [hunk 2]
```diff
@@ -36,19 +42,20 @@ node_port(NodeName, erlpmd) ->
...
 		[[NodeName]] -> remove_node(NodeName, erlpmd);
 		_ -> ok
 	end.
 
 remove_node(NodeName, erlpmd) ->
-    ets:delete(NodeName).
+    true = ets:delete(erlpmd, NodeName),
+    {ok, erlpmd}.
```
Comment (sanmiguel) [inline src/erlpmd_ets.erl:61]: This was definitely a bug.

File: src/erlpmd_ets.erl (hunk 1 above)
Comment (sanmiguel) [inline src/erlpmd_ets.erl:54]: Sounds about right - I sorta threw that bit together.

File: src/erlpmd_ets.erl (hunk 2 above)
Comment (sanmiguel) [inline src/erlpmd_ets.erl:61]: I'm going to fix this delete return type bug in mtc-develop before I merge it into master

**Comment by sanmiguel** [comment]
I'm wondering if this can't be pushed up a level and used in `erlpmd` instead. Leave it with me for now?

**Comment by puzza007** [comment]
:+1:

**Pull request closed without merging**